flask_static_digest/cli.py

```diff
@@ -1,21 +1,83 @@
-import click
+from dataclasses import dataclass
+from warnings import warn
 
+import click
 from flask import current_app
 from flask.cli import with_appcontext
 
-from flask_static_digest.digester import compile as _compile
 from flask_static_digest.digester import clean as _clean
+from flask_static_digest.digester import compile as _compile
+
+
+@dataclass
+class CompressionAlgo:
+    gzip: bool = False
+    brotli: bool = False
 
 
 @click.group()
-def digest():
-    """md5 tag and gzip static files."""
-    pass
+@click.pass_context
+@with_appcontext
+def digest(ctx):
+    """md5 tag and compress static files."""
+
+    ctx.obj = comp_algo = CompressionAlgo()
+    flask_static_comp = current_app.config.get(
+        "FLASK_STATIC_COMPRESSION", None
+    )
+    flask_static_digest_gzip = current_app.config.get(
+        "FLASK_STATIC_DIGEST_GZIP_FILES", None
+    )
+    if flask_static_comp is not None and flask_static_digest_gzip is not None:
+        click.echo(
+            "Error: Flask-Static-Digest options FLASK_STATIC_DIGEST_GZIP_FILES"
+            " (depreciated) and FLASK_STATIC_COMPRESSION are set. Delete"
+            " depreciated FLASK_STATIC_DIGEST_GZIP_FILES.",
+            err=True,
+        )
+        exit(-1)
+
+    if flask_static_digest_gzip is not None:
```

Comment:
I think we might be able to handle this deprecation notice in `init_app` in the `__init__.py` file. Conceptually the deprecation is at the extension level, not the digest level.

Also related to the `init_app` function, we'll want to define `app.config.setdefault("FLASK_STATIC_DIGEST_BLACKLIST_COMPRESSION", [])`

Reply:
For setting the default in the `init_app` function, I think we can do something even better to handle the soon to be deprecated gzip config option.

Something like this:

```py
app.config.setdefault("FLASK_STATIC_DIGEST_GZIP_FILES", True)
app.config.setdefault("FLASK_STATIC_DIGEST_COMPRESSION", ["gzip"])

if app.config.get("FLASK_STATIC_DIGEST_GZIP_FILES"):
    app.config["FLASK_STATIC_DIGEST_COMPRESSION"].append("gzip")
```

This way our implementation only ever needs to worry about the compression value and that if condition can be deleted once 0.5 is released. The implementation also converts the list to a set so the duplicate gzip values will be made unique.